// @angular-mdl/core/components/common/index.js
import { NgModule } from '@angular/core';
import { AppendViewContainerRefDirective } from './append-view-container-ref-directive';
import { Animations, NativeWebAnimations, NoopWebAnimations } from './animations';
function isWebAnimationsSupported() {
    return typeof Element !== 'undefined' && typeof Element.prototype['animate'] === 'function';
}
export function instantiateSupportedAnimationDriver() {
    /* istanbul ignore next */
    if (isWebAnimationsSupported()) {
        return new NativeWebAnimations();
    }
    /* istanbul ignore next */
    return new NoopWebAnimations();
}
var MdlCommonsModule = (function () {
    function MdlCommonsModule() {
    }
    return MdlCommonsModule;
}());
export { MdlCommonsModule };
MdlCommonsModule.decorators = [
    { type: NgModule, args: [{
                imports: [],
                exports: [AppendViewContainerRefDirective],
                declarations: [AppendViewContainerRefDirective],
                providers: [
                    { provide: Animations, useFactory: instantiateSupportedAnimationDriver }
                ]
            },] },
];
/** @nocollapse */
MdlCommonsModule.ctorParameters = function () { return []; };
